{
  "url": "https://api.github.com/repos/quarkusio/quarkus/issues/30512",
  "repository_url": "https://api.github.com/repos/quarkusio/quarkus",
  "labels_url": "https://api.github.com/repos/quarkusio/quarkus/issues/30512/labels{/name}",
  "comments_url": "https://api.github.com/repos/quarkusio/quarkus/issues/30512/comments",
  "events_url": "https://api.github.com/repos/quarkusio/quarkus/issues/30512/events",
  "html_url": "https://github.com/quarkusio/quarkus/issues/30512",
  "id": 1551143762,
  "node_id": "I_kwDOCFbutM5cdJNS",
  "number": 30512,
  "title": "Cache key resolution based on reflection calls",
  "labels": [
    {
      "id": 1808747697,
      "node_id": "MDU6TGFiZWwxODA4NzQ3Njk3",
      "url": "https://api.github.com/repos/quarkusio/quarkus/labels/area/cache",
      "name": "area/cache",
      "color": "0366d6",
      "default": false,
      "description": ""
    }
  ],
  "state": "open",
  "locked": false,
  "milestone": null,
  "comments": 3,
  "created_at": "2023-01-20T16:34:20Z",
  "updated_at": "2023-01-23T08:34:41Z",
  "closed_at": null,
  "active_lock_reason": null,
  "body": "Hello,\r\nI'm trying to cache a resteasy http client which returns a jwt token.\r\nI get this WARN message when I call the cached method : \"Cache key resolution based on reflection calls. Please create a GitHub issue in the Quarkus repository, the maintainers might be able to improve your application performance.\" (see `io.quarkus.cache.runtime.CacheInterceptor`).\r\n\r\nHere is the code I use:\r\n\r\n```java\r\n@Path(\"/api\")\r\npublic interface MyHttpClient {\r\n    @POST\r\n    @Path(\"/authorization\")\r\n    @Produces(MediaType.APPLICATION_JSON)\r\n    @CacheResult(cacheName = \"token-cache\")\r\n    String getToken(Credentials credentials);\r\n}\r\n```\r\n\r\nHere is how the cache is configured in my `application.properties`:\r\n\r\n```properties\r\nquarkus.cache.caffeine.token-cache.expire-after-write=PT600s\r\n```\r\nThe cache works great but I always have the WARN message when I call the method.  \r\n  \r\nWhat do you think, is it a code/configuration issue or a real quarkus bug?  \r\n  \r\nThanks!",
  "reactions": {
    "url": "https://api.github.com/repos/quarkusio/quarkus/issues/30512/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/quarkusio/quarkus/issues/30512/timeline",
  "performed_via_github_app": null,
  "state_reason": null
}
